### DISCUSSION TITLE
Inner logic of IF - AND - OR

### DISCUSSION THREAD

**Discussion by dasToshy** [WarmUpTill/SceneSwitcher#628] [Q&A]
Good evening.

Despite many tests, I still don't quite understand the inner logic of the IF - AND - OR connections.

How do IF AND and OR work together?
Depending on the order in which I set them up, the conditions are partly used and partly not, without me recognizing the context in which this is happening.

I use 1x IF and 2 x AND and 2 x OR for the test. Just move them in order. Two tests in the video (see highlight).

Sometimes 1 AND is not asked (although I would guess it should be used) and in a different order NO OR is asked but ONLY the "AND".

An explanation of the exact way of working would be helpful.
thank you very much

And as always, excuse my bad english.
Greetings Tushy

https://user-images.githubusercontent.com/109924400/206881259-ed479617-421f-4e53-8a1f-3399457fb03e.mp4

**Comment by dasToshy** [comment]
I'm trying to combine several (at least 2) AND conditions with OR conditions.

At least 2 AND conditions should apply and 1 of the OR conditions.
So I could make complex query. If necessary, outsource further queries to additional macros.
I can't get the joint use of "AND" AND "OR" to work consistently.

In the following video, the OR conditions are completely ignored. (If I move the OR conditions up to the "IF" and "And" down, the "And" conditions are not fully observed [see first video])


https://user-images.githubusercontent.com/109924400/206881941-4157bb6b-141f-4f80-80f9-d13670e5280f.mp4

Maybe one query always relates to the previous one, I just don't know and I test a lot. However, this is not easy and not certain to find out with a pure test.

**Comment by WarmUpTill** [comment]
The conditions are executed from top to bottom.
So your first example would result in the following logical expression:

`(((Date AND Variable) AND Variable) OR Variable) OR Variable`

The conditions are always checked even if it might not change the outcome of the final result.

Does that clear things up?
